text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+The process of `signing` involves generating a signature for a crate, using a private key, and attaching that signature when publishing a crate. The process of `verification` involves downloading a crate (using cargo) and using a _trusted_ public key to verify the authenticity (the signer) and integrity (the contents) of a crate.
+
+To simplify key management for users, cargo supports using [Sigstore](https://sigstore.dev) as a way to generate ephemeral keys based on an identity, and as an immutable log that can be monitored and audited in the event of a compromise. By default, a publicly available instance ensures that you do not need to manage this infrastructure yourself. If you are running your own crate registry, Sigstore instance and identify provider, you can configure cargo to use that.
+
+Verification is accomplished through an API interaction with the relevant Rekor service. This occurs at artifact submission time to the relevant crates registry and is also how the consumer will verify the artifact whether manually or in a run-time scenario.
+
+## Implementation sequencing
+
+## Phase 1
+
+* Establish the required metadata structures and modifications to crates.io to accommodate the Sigstore signing process.
+* Modify cargo to add the Sigstore specific capabilities for signing and verification, and configuration options to require signed artifacts from crates.io during dependency pulls, etc.
+* Modify crates.io to accept the additional metadata attributes from the signing operation.
+* Provide inspection / verification capabilities in the crates.io UI. for Sigstore signed crates
+
+## Phase 2
+
+* TUF (The Update Framework) - Phase 2 TUF AuthZ Architecture
+   * The TUF protocol addresses a set of defined attacks and threat models specific to software distribution systems. It has the capability of providing a strong AuthZ implementation for any OSS project with respect to control of the ability to generate an artifact. For example, it is used by the Sigstore project itself to instantiate and manage the Sigstore Public Root. See: [public Sigstore instance](https://blog.sigstore.dev/a-new-kind-of-trust-root-f11eeeed92ef). This article does a good job of showing the options.
+* Make Sigstore signed crates mandatory.
+* Start marking legacy / unmaintained unsigned crates as unsafe
+* Add  metadata attributes to allow for identification of build source,  i.e. “certified” system such as GitHub actions. This could also take the form of using Sigstore’s organic countersigning capability.
+
+## Configuration changes to cargo
+
+By default, cargo will use the publicly available Sigstore instances together with GitHub as the identity provider. To override (if using a different registry than crates.io), the `$HOME/.cargo/sigstore.toml` file can be configured with the location of Sigstore and identity services:
```

Comment:
Why would anyone want to do this? Without this information, this section is useless. Additionally, mentions of this are absent from the motivation, making it unclear why this would be preferred over just forcing everyone on the same ones associated with a registry.

Reply:
I agree a justification is needed here. 

On its own, I wonder about the value/benefit of limiting the "default" IdP to GitHub -- as others have pointed out it raises both "lock-in" and "lock-out" concerns, which is why Sigstore supports several IdPs by default (and without any client-side configuration).